src/resmom/mom_main.c

```diff
@@ -8309,15 +8309,39 @@ main(int argc, char *argv[])
 #if defined(RLIM64_INFINITY)
 	{
 		struct rlimit64 rlimit;
+		int curerror;
 
 		rlimit.rlim_cur = RLIM64_INFINITY;
 		rlimit.rlim_max = RLIM64_INFINITY;
 
 		(void)setrlimit64(RLIMIT_CPU,   &rlimit);
 		(void)setrlimit64(RLIMIT_FSIZE, &rlimit);
 		(void)setrlimit64(RLIMIT_DATA,  &rlimit);
-		(void)getrlimit64(RLIMIT_STACK, &orig_stack_size); /* get for later */
-		(void)setrlimit64(RLIMIT_STACK, &rlimit);
+		if (getrlimit64(RLIMIT_STACK, &orig_stack_size) != -1) {
+			if((orig_stack_size.rlim_cur != RLIM_INFINITY) && (orig_stack_size.rlim_cur < MIN_STACK_LIMIT)) {
+				rlimit.rlim_cur = MIN_STACK_LIMIT;
+				rlimit.rlim_max = MIN_STACK_LIMIT;
```

Comment:
why do we set rlim_max to MIN_STACK_LIMIT as well?

Reply:
rlim_max is a hard limit (ceiling for rlim_cur) & This hard limit acts as a ceiling for the soft limit. 
 A privileged process may make arbitrary changes to either limit value. 

Reason is just to make the stack size fixed to "MIN_STACK_LIMIT" and not allowing to use beyond this value.
 
Info taken from GNU/Linux man page of "setrlimit", their sample code insist to do so. [man page](http://man7.org/linux/man-pages/man2/getrlimit.2.html)